src/rgw/rgw_admin.cc

```diff
@@ -1030,7 +1030,11 @@ void set_quota_info(RGWQuotaInfo& quota, int opt_cmd, int64_t max_size, int64_t
 
     case OPT_QUOTA_SET:
       if (have_max_objects) {
-        quota.max_objects = max_objects;
+        if (max_objects < 0) {
+          quota.max_objects = -1;
+        } else {
+          quota.max_objects = max_objects;
+        }
```

Comment:
is this change really needed? the quota checks just use >= 0, so we're not really treating -1 as a special value